whd/protocol.go

```diff
@@ -51,17 +52,17 @@ func (s *SDPCMHeader) Put(order binary.ByteOrder, dst []byte) {
 }
 
 func (s *SDPCMHeader) Parse(packet []byte) (payload []byte, err error) {
-	if len(packet) < SDPCM_HEADER_LEN {
-		err = errors.New("packet shorter than sdpcm hdr, len=", strconv.Itoa(len(packet)))
+	if len(packet) < SDPCM_HEADER_LEN+int(s.Size) {
```

Comment:
Adding +int(s.Size) is different than what the ref does.  This first check is just to see if the packet has enough to cover the hdr.

The full size check is done lower, around line 65.

Reply:
fixed! I think